llvm/unittests/ADT/SetOperationsTest.cpp

```diff
@@ -273,4 +273,15 @@ TEST(SetOperationsTest, SetIsSubset) {
   EXPECT_FALSE(set_is_subset(Set1, Set2));
 }
 
+TEST(SetOperationsTest, SetIntersects) {
+  std::set<int> Set1 = {1, 2, 3, 4};
+  std::set<int> Set2 = {3, 4, 5};
+  EXPECT_TRUE(set_intersects(Set1, Set2));
+  EXPECT_TRUE(set_intersects(Set2, Set1));
+
+  Set2 = {5, 6, 7};
+  EXPECT_FALSE(set_intersects(Set1, Set2));
+  EXPECT_FALSE(set_intersects(Set2, Set1));
+}
```

Comment:
Could you also add a testcase with empty sets (the usual source of edge cases)?

Reply:
done